mono/mini/aot-compiler.c

```diff
@@ -9814,11 +9814,13 @@ emit_llvm_file (MonoAotCompile *acfg)
 		// FIXME: This doesn't work yet
 		opts = g_strdup ("");
 	} else {
-		opts = g_strdup ("-O2 -disable-tail-calls -place-safepoints -spp-all-backedges");
+		opts = g_strdup ("-disable-tail-calls -place-safepoints -spp-all-backedges");
 	}
 
 	if (acfg->aot_opts.llvm_opts) {
```

Comment:
Doesn't this disable -O2 even if llvm_opts was used to specify some conflicting option (i.e. an option that is not an optimization level)? Is that the desired behavior?

Reply:
yes, but I don't see a better solution - we must allow users to specify the opt level because `-Os` actually does show nice size improvements (and currently it's not possible to turn it on basically). As an alternative - a separate flag but we already have tons of them...

Reply:
maybe do not use any default options at all if `llvmopts` is used? It's kinda weird to omit `-O2`, but not the other default options.

Reply:
@lewurm heh 🙂 it's actually a feature - it used to be the way you described - we had to copy all these default **mandatory** options. I think the rest 3 options (and more in case of `llc`) must be there always.

Reply:
I have not been explicitly setting it in the perf lab.

I don't have a better solution, but we might want to also update the man page, because it currently says that using llvmopts is strictly additive.